### PR TITLE
Bring back lost 'follow' (redirects) request option

### PR DESCRIPTION
**Pull request by robingustafsson**
I needed this functionality and then realized we seem to have lost it in the Otto -> Goja migration, as it was previously implemented, see https://github.com/loadimpact/k6/issues/232.

Usage:
```js
http.get("https://httpbin.org/redirect/1", {follow: false});
check(res, {
    "has status 302": (r) => r.status === 302,
    "has non-redirected URL": (r) => r.url === "https://httpbin.org/redirect/1"
})
```

### CODE REVIEW

**Comment by liclac** [comment]
I think this option would do better as an override for the number of redirects.

**Comment by robingustafsson** [comment]
@liclac Ok, sounds reasonable. And a `maxRedirects: 0` would return last response (`return http.ErrUseLastResponse` from `CheckRedirect` in Go) rather than throwing the "max redirects..." error? Because that's a real use case, that you want to interfere with redirects.

**Comment by liclac** [comment]
What if `follow` could be either a number or a bool, and if it's false that happens? But at that point, it may be better to just have two different options...

**Comment by luxifer** [comment]
Hi guys, this is a serious BC break and I really need k6 to not fail when there is a redirect...

**Comment by luxifer** [comment]
With `maxRedirects: 0` in the options, the current respons status is 0, instead of the real http status code (should be 302 in my case).

**Comment by robingustafsson** [comment]
@luxifer Do you mean removing `follow: true|false` would be a BC break, or are you saying that changing the behavior of `maxRedirects: 0` is the BC break? In the latter case would you rather have the current behavior of an error being logged and status being 0, than having the first redirected response with 3xx status returned?

I'm thinking the best way forward might be to keep the `maxRedirects` behavior for now (keep it as a separate discussion whether to break that compatibility), and just reintroducing the `follow: true|false` as it's implemented in this PR right now. I'm also fine with implementing the `follow: true|false|number` version, although I'm a bit hesitant to having `follow` doubling as a per-request override for the global `maxRedirects` option under a different name.

@liclac Thoughts?

**Comment by luxifer** [comment]
That's exactly what I meant. The current behavior pour throwing an error when `mawRedirects` is 0 and the actual response code is 3xx is not good. I would like the `follow` option on http requests to be reintroduced thot that I can check that I am currently redirected.

Thanks!

**Comment by robingustafsson** [comment]
Ping @liclac

**Comment by liclac** [comment]
What about this:

* Default to following 10 redirects
* Use `{ redirects: 5 }`, etc. to change this
* Return the last response, but **log a warning** if the `redirects` option isn't specified to prevent confusion ("why is my check failing??")

**Comment by robingustafsson** [comment]
@liclac Ok, that sounds like a good idea to me. I'll change the implementation.

**Comment by robingustafsson** [comment]
@liclac Now updated.

File: js/modules/k6/http/http.go [hunk 1]
```diff
@@ -189,9 +196,18 @@ func (*HTTP) request(ctx context.Context, rt *goja.Runtime, state *common.State,
...
+			max := state.Options.MaxRedirects.Int64
+			if redirects > -1 {
+				max = redirects
+			}
+			if len(via) > int(max) {
+				if redirects == -1 {
+					state.Logger.WithFields(log.Fields{
+						"error": fmt.Sprintf("stopped after %d redirects (controlled via option \"maxRedirects\")", max),
+						"url":   via[0].URL.String(),
+					}).Warn("Redirect Limit")
```
Comment (liclac) [inline js/modules/k6/http/http.go]: Make this something like "Possible redirect loop, {{ via[0].Status }} response returned; pass `{ redirects: n }` to silence this". It doesn't matter if this is long, it should only be printed very rarely to make sure the user isn't accidentally doing the wrong thing.

**Comment by robingustafsson** [comment]
@liclac Requested changes have been made.

**Pull request merged**